homeassistant/components/soundtouch/media_player.py

```diff
@@ -50,137 +49,58 @@
-        master_device_id = service.data.get("master")
-        slaves_ids = service.data.get("slaves")
-        slaves = []
-        if slaves_ids:
-            slaves = [
-                device
-                for device in hass.data[DATA_SOUNDTOUCH]
-                if device.entity_id in slaves_ids
-            ]
-
-        master = next(
-            iter(
-                [
-                    device
-                    for device in hass.data[DATA_SOUNDTOUCH]
-                    if device.entity_id == master_device_id
-                ]
-            ),
-            None,
-        )
-
-        if master is None:
-            _LOGGER.warning(
-                "Unable to find master with entity_id: %s", str(master_device_id)
-            )
-            return
-
-        if service.service == SERVICE_PLAY_EVERYWHERE:
-            slaves = [
-                d for d in hass.data[DATA_SOUNDTOUCH] if d.entity_id != master_device_id
-            ]
-            master.create_zone(slaves)
-        elif service.service == SERVICE_CREATE_ZONE:
-            master.create_zone(slaves)
-        elif service.service == SERVICE_REMOVE_ZONE_SLAVE:
-            master.remove_zone_slave(slaves)
-        elif service.service == SERVICE_ADD_ZONE_SLAVE:
-            master.add_zone_slave(slaves)
-
-    hass.services.register(
-        DOMAIN,
-        SERVICE_PLAY_EVERYWHERE,
-        service_handle,
-        schema=SOUNDTOUCH_PLAY_EVERYWHERE,
-    )
-    hass.services.register(
-        DOMAIN,
-        SERVICE_CREATE_ZONE,
-        service_handle,
-        schema=SOUNDTOUCH_CREATE_ZONE_SCHEMA,
+    """Set up the Bose SoundTouch platform."""
+    _LOGGER.warning(
+        "Configuration of the Bose SoundTouch platform in YAML is deprecated and will be "
+        "removed in a future release; Your existing configuration "
+        "has been imported into the UI automatically and can be safely removed "
+        "from your configuration.yaml file"
     )
-    hass.services.register(
-        DOMAIN,
-        SERVICE_REMOVE_ZONE_SLAVE,
-        service_handle,
-        schema=SOUNDTOUCH_REMOVE_ZONE_SCHEMA,
-    )
-    hass.services.register(
-        DOMAIN,
-        SERVICE_ADD_ZONE_SLAVE,
-        service_handle,
-        schema=SOUNDTOUCH_ADD_ZONE_SCHEMA,
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
     )
 
 
-class SoundTouchDevice(MediaPlayerEntity):
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Bose SoundTouch media player based on a config entry."""
+    device = hass.data[DOMAIN][entry.entry_id].device
+    media_player = SoundTouchMediaPlayer(entry.data[CONF_NAME], device)
+
+    async_add_entities([media_player], True)
+
+    hass.data[DOMAIN][entry.entry_id].media_player = media_player
```

Comment:
We shouldn't let other parts of the integration directly call methods on this media player. Once you switch to registering services via `entity_platform` you don't need to anyway 👍

Reply:
Correct. I planned on removing this once the join/unjoin services are in place